transform.ml

```diff
@@ -1,7 +1,7 @@
 open Instr
 
 (* types of transformations *)
-type transform_instructions = instruction_stream -> instruction_stream option
+type transform_instructions = analysis_input -> analysis_input option
```

Comment:
Ah so the problem is that if transform_instructions is not transitive I cannot use the same operator for chaining instruction transformations and program and function transformations.

I have changed it now (because I agree that outputting the formals is dumb). To fix the operator problem I have introduced two new stream operators. (==>) makes a transform_instruction transitive and (<==) turns it back into a normal transform_instruction. Now to chain instruction transformation you have to:

`(<==) ((==>) t1 &&& (==>) t2)`